test/e2e/sched/sched_test.go

```diff
@@ -39,11 +42,26 @@ import (
 
 var _ = Describe("[Scheduler] imageReplacement", func() {
 	var initialized bool
+	nroSchedObj := &nropv1.NUMAResourcesScheduler{}
 	BeforeEach(func() {
 		if !initialized {
 			Expect(e2eclient.ClientsEnabled).To(BeTrue(), "failed to create runtime-controller client")
 		}
 		initialized = true
+		nroSchedKey := objects.NROSchedObjectKey()
+		Expect(e2eclient.Client.Get(context.TODO(), nroSchedKey, nroSchedObj)).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroSchedKey.String())
+
+		DeferCleanup(func() {
+			Expect(e2eclient.Client.Get(context.TODO(), nroSchedKey, nroSchedObj)).ToNot(HaveOccurred())
+			nroSchedObj.Spec = objects.TestNROScheduler().Spec
+			Expect(e2eclient.Client.Update(context.TODO(), nroSchedObj)).ToNot(HaveOccurred())
```

Comment:
this can fail. The `Get/Revert/Update` should be wrapped in its own `Eventually`. We hit this already multiple time